### PR TITLE
DEVPROD-9871 Upgrade go driver to v2 beta

### PR DESCRIPTION
**Pull request by ZackarySantana**
[DEVPROD-9871](https://jira.mongodb.org/browse/DEVPROD-9871)

There's a list of migration details in the ticket's linked google doc V2 Migration guide.

### CODE REVIEW

**Review by malikchaya2 — APPROVED** [review]

File: db/interface.go [hunk 1]
```diff
@@ -1,7 +1,5 @@
 package db
 
-import "time"
```
Comment (malikchaya2) [inline db/interface.go:3]: nit: is this change needed?

Comment (ZackarySantana) [reply db/interface.go:3]: Yes, the driver no longer exposes a max time or timeout. To do the same behavior, you need to do a context with a timeout. They document it as it is deprecated and I added some notes here: https://github.com/mongodb/anser/pull/92/files#diff-663485385d1519cc72821f4b6ca656c52b0d1b4bee4d414df910f2a01ddfd9b3R483 and similar functions.

Comment (ZackarySantana) [reply db/interface.go:3]: I think it's fair to straight up remove it and not keep supporting it similar to the mongodb driver did.
